zcash_client_backend/src/scanning.rs

```diff
@@ -141,21 +141,46 @@ pub enum ScanError {
     },
 }
 
+impl ScanError {
+    /// Returns whether this error is the result of a failed continuity check
+    pub fn is_continuity_error(&self) -> bool {
+        use ScanError::*;
+        match self {
+            PrevHashMismatch { .. } => true,
+            BlockHeightDiscontinuity { .. } => true,
+            TreeSizeMismatch { .. } => true,
+            TreeSizeUnknown { .. } => false,
+        }
+    }
+
+    /// Returns the block height at which the scan error occurred
+    pub fn at_height(&self) -> BlockHeight {
+        use ScanError::*;
+        match self {
+            PrevHashMismatch { at_height } => *at_height,
+            BlockHeightDiscontinuity { new_height, .. } => *new_height,
+            TreeSizeMismatch { at_height, .. } => *at_height,
+            TreeSizeUnknown { at_height, .. } => *at_height,
+        }
+    }
+}
+
 impl fmt::Display for ScanError {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use ScanError::*;
         match &self {
-            ScanError::PrevHashMismatch { at_height } => write!(
+            PrevHashMismatch { at_height } => write!(
                 f,
                 "The parent hash of proposed block does not correspond to the block hash at height {}.",
                 at_height
             ),
-            ScanError::BlockHeightDiscontinuity { prev_height, new_height } => {
+            BlockHeightDiscontinuity { prev_height, new_height } => {
                 write!(f, "Block height discontinuity at height {}; next height is : {}", prev_height, new_height)
```

Comment:
The message here uses "at height" to refer to `prev_height`, but the `at_height()` method returns `new_height`. That is potentially confusing.